File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/issue/IssueType.java
package org.ifinalframework.jetbrains.plugins.aio.issue;


import com.intellij.openapi.util.IconLoader;

import org.ifinalframework.jetbrains.plugins.aio.icon.Icons;

import javax.swing.Icon;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;

/**
 * IssueType
 *
 * @author iimik
 * @since 1.6.0
 **/
@Getter
@RequiredArgsConstructor
public enum IssueType {
    issue(IconLoader.getIcon("/assets/git.svg", Icons.class)),
    jira(IconLoader.getIcon("/assets/jira.svg", Icons.class));

    private final Icon icon;


    public static IssueType of(final String name) {
        return Arrays.stream(values())
                .filter(type -> type.name().equalsIgnoreCase(name))
                .findFirst()
                .orElse(null);
    }
}

File: idea-plugin/src/main/kotlin/org/ifinalframework/jetbrains/plugins/aio/issue/IssueType.kt
package org.ifinalframework.jetbrains.plugins.aio.issue

import com.intellij.openapi.util.IconLoader
import com.intellij.util.containers.stream
import org.ifinalframework.jetbrains.plugins.aio.icon.Icons
import javax.swing.Icon


/**
 * IssueType
 *
 * @author iimik
 * @since 0.0.1
 **/
enum class IssueType(
    val icon: Icon,
) {
    ISSUE(IconLoader.getIcon("/assets/git.svg", Icons::class.java)),
    JIRA(IconLoader.getIcon("/assets/git.svg", Icons::class.java));

    companion object {
        fun ofNullable(name: String?): IssueType? {
            return values().stream()
                .filter { v -> v.name.equals(name, ignoreCase = true) }
                .findFirst()
                .orElse(null)
        }
    }
}